### PR TITLE
feat(api): allow admins to create and edit comments

### PR DESCRIPTION
**Pull request by Itrulia**
```graphql
mutation createComment(
	$itemID: ID!
	$itemType: CommentItemType!
	$text: RichText!
	$tagIds: [ID!]
) {
	createComment(
		itemID: $itemID
		itemType: $itemType
		text: $text
		tagIds: $tagIds
	) {
		id
		revisions {
			text
		}
		tags {
			id
			tag
		}
	}
}

// vars
{
  "itemID": "cl7673h4u1158mspx2rpytjvk",
  "itemType": "Article",
  "tagIds": ["cl7672l4e0839mspxi0bfzhpc"],
  "text": [
    {
      "type": "paragraph",
      "children": [
        {
          "text": "Haha"
        }
      ]
    }
  ]
}
```

```graphql
mutation updateComment($id: ID!, $text: RichText, $tagIds: [ID!]) {
	updateComment(id: $id, text: $text, tagIds: $tagIds) {
		id
		revisions {
			text
		}
		tags {
			id
			tag
		}
	}
}

// vars
{
  "id":"cl7679xnq02287fpxggyvj0n4",
  "tagIds": ["cl767aneo05387fpx76u8lcwy", "cl7672l4e0839mspxi0bfzhpc"],
  "text": [
    {
      "type": "paragraph",
      "children": [
        {
          "text": "Haha2"
        }
      ]
    }
  ]
}
```

### CODE REVIEW

File: packages/api/src/graphql/comment/comment.private-mutation.ts [hunk 1]
```diff
@@ -19,3 +25,85 @@ export const takeActionOnComment = (
...
+    data: {
+      revisions: text
+        ? {
+            create: {
+              text
+            }
+          }
+        : undefined,
+      tags: tagIds
+        ? {
```
Comment (michael-scheurer) [inline packages/api/src/graphql/comment/comment.private-mutation.ts:50]: quite unreadable. I would refactor that in a separate variable before commentClient.update() method.

File: packages/api/src/graphql/comment/comment.private-mutation.ts [hunk 2]
```diff
@@ -19,3 +25,85 @@ export const takeActionOnComment = (
...
+            connectOrCreate: tagIds.map(tagId => ({
```
Comment (michael-scheurer) [inline packages/api/src/graphql/comment/comment.private-mutation.ts:51]: What if, CAN_CREATE_TAG permission is missing?

File: packages/api/src/graphql/comment/comment.private-mutation.ts [hunk 3]
```diff
@@ -19,3 +25,85 @@ export const takeActionOnComment = (
...
+      authorType: CommentAuthorType.team,
+      itemID: itemId,
+      itemType,
+      revisions: {
+        create: {
+          text
+        }
+      },
+      tags: {
+        create: tagIds?.map(tagId => ({
```
Comment (michael-scheurer) [inline packages/api/src/graphql/comment/comment.private-mutation.ts:100]: What if, CAN_CREATE_TAG permission is missing?

**Review by michael-scheurer — CHANGES_REQUESTED** [review]

File: packages/api/src/graphql/comment/comment.private-mutation.ts (hunk 2 above)
Comment (Itrulia) [reply packages/api/src/graphql/comment/comment.private-mutation.ts:51]: This is just for the m to m table, so it adds a record to bind a comment to a tag.

**Pull request merged**